### ISSUE TITLE
Selected value will be reset while the Select control is under FormItem 

### ISSUE THREAD

**Issue by ldsenow** [ant-design-blazor/ant-design-blazor#3355]
Hi,

I am having issue after upgrading the version from 0.13.2 to 0.15.3. 

Selected value will be reset while the Select control is under FormItem. I tracked down to the line which is causing the issue which is line 358 in FormItem.razor.cs

![image](https://github.com/ant-design-blazor/ant-design-blazor/assets/177045/3804f00c-30ea-4d98-90a8-489256b19f74)

Once I commented out the line, all the bindings work fine. control.OnValidated(_validationMessages); is actually have raised the StateHasChanged as well, not quite sure if it is about duplicate of the StateHasChanged calls.

Looks like it happens since 0.13.3. Hope it will be taken care by someone shortly. Meanwhile, I am downgrading to 0.13.2

**Comment by ElderJames** [comment]
Thank for contacting us @ldsenow . We need to raised `StateHasChanged` for showing the validation message for formitem. And it work correctly with following code:

```
@using System.ComponentModel.DataAnnotations;
@using System.Text.Json;
@using System.ComponentModel

    <Form Model="@model"
          OnFinish="OnFinish"
          OnFinishFailed="OnFinishFailed"
          LabelColSpan="8"
          WrapperColSpan="16">
        <FormItem Label="Username">
        <Select @bind-Value="@context.Username"
                Style="width: 100%;"
                TItemValue="string"
                TItem="string"
                >
            <SelectOptions>
                <SelectOption TItemValue="string" TItem="string" Value="@("lucy")" Label="Lucy" />
                <SelectOption TItemValue="string" TItem="string" Value="@("james")" Label="Jucy" />
            </SelectOptions>
        </Select>
        </FormItem>
        <FormItem Label="Password">
            <InputPassword @bind-Value="@context.Password" />
        </FormItem>
        <FormItem WrapperColOffset="8" WrapperColSpan="16">
            <Checkbox @bind-Value="context.RememberMe">Remember me</Checkbox>
        </FormItem>
        <FormItem WrapperColOffset="8" WrapperColSpan="16">
            <Button Type="@ButtonType.Primary" HtmlType="submit">
                Submit
            </Button>
        </FormItem>
    </Form>
@code
{
    public class Model
    {
        [Required, DisplayName("User Name")]
        public string Username { get; set; }
        [Required]
        public string Password { get; set; }
        public bool RememberMe { get; set; } = true;
    }

    protected override async Task OnInitializedAsync()
    {
        await Task.Delay(1000);
        model.Username = "james";
    }

    private Model model = new Model();

    private void OnFinish(EditContext editContext)
    {
        Console.WriteLine($"Success:{JsonSerializer.Serialize(model)}");
    }

    private void OnFinishFailed(EditContext editContext)
    {
        Console.WriteLine($"Failed:{JsonSerializer.Serialize(model)}");
    }
}
```

![image](https://github.com/ant-design-blazor/ant-design-blazor/assets/7550366/748b08d8-5b52-4a8d-a783-79d6c3f2c31f)

**Comment by ldsenow** [comment]
Hi @ElderJames , thanks for looking into that. I found the issue is when I set the ValidateOnChange="true", the issue happens.

**Comment by ElderJames** [comment]
When I debug it, I find that this only happens when FormCustomSelect wraps the Select component, and that using Select alone is fine.
I think the two-way binding is not binding the external Value properly, causing blazor to assign a null value to the value when StateHasChanged is called.

**Comment by ldsenow** [comment]
Unfortunately, I have a few custom components wrapped in that way, and it worked up until I recently tried to update the package version. I did a simple console log and found the binding is fine, the value is just get reset. Not sure why the change event is being triggered a few times. I tracked down to the StateHasChanged function is causing reset of the value and it is set back to the previous selected value. So I think it is a bug in the component not in the blazor framework

**Comment by ldsenow** [comment]
I just went a bit further. The issue will be triggered when ValidateOnChange option is turned on in the Form component.
@ElderJames

**Comment by ElderJames** [comment]
I've seen it, but I haven't thought of a solution yet.

**Comment by ldsenow** [comment]
Can i help investigate it if no one is on it? 
It would be good you @ElderJames  can get me some tips/areas to look at, I will try my best.

**Comment by ElderJames** [comment]
It's very kind of you to help @ldsenow 
I found out through debugging that it was the refresh of the validation information that caused select to remove the option, which involved a problem with the loop call.

**Comment by ldsenow** [comment]
@ElderJames I will give it a go and thanks for the tip. TBH, I will avoid calling StateHasChanged, it feels like a black magic.

**Comment by ElderJames** [comment]
Also need to ensure the validation still works.

**Comment by ldsenow** [comment]
@ElderJames  Have worked out to remove the redundant call OnValueChanged on the OnParametersSet of Select.cs, it resolves the issue

**Comment by ElderJames** [comment]
That's great @ldsenow !

**Comment by ldsenow** [comment]
I know the issue is closed. However, the issue comes back again from 0.17.4+. I guess the fix wasn't a good fix and might have caused other issues. TBH, there are too many StateHasChange / events around the Select control, it is a bit hard to follow how the flow is along the way.

**Comment by ElderJames** [comment]
Hello @ldsenow , maybe we can fix it again 😂
